core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -95,13 +95,91 @@ oppia.directive('topNavigationBar', [function() {
           SidebarStatusService.closeSidebar();
+
+          // If the window is resized larger, try displaying the hidden elements
+          if (currentWindowWidth < windowDimensionsService.getWidth()) {
+            for (element in $scope.navElementsVisibilityStatus) {
+              if (!$scope.navElementsVisibilityStatus[element]) {
+                $scope.navElementsVisibilityStatus[element] = true;
+              }
+            }
+          }
+          currentWindowWidth = windowDimensionsService.getWidth();
+          truncateNavbarDebounced();
         });
 
+        /**
+         * Checks if i18n has been run.
+         * If i18n has not yet run, the <a> and <span> tags will have
+         * no text content, so their innerText.length value will be 0.
+         * @returns {boolean}
+         */
+        var checkIfI18NCompleted = function() {
+          var i18nCompleted = true;
+          document.querySelectorAll('.oppia-navbar-tabs a[translate], ' +
+            '.oppia-navbar-tabs span[translate]').forEach(function(element) {
+            if (element.innerText.length === 0) {
+              i18nCompleted = false;
+              return false;
+            }
+          });
+          return i18nCompleted;
+        };
+
+        /**
+         * Checks if window is >768px and i18n is completed, then checks
+         * for overflow. If overflow is detected hides the least important
+         * tab and then calls itself again after a 50ms delay.
+         */
+        var truncateNavbar = function() {
+          // If the window is narrow, the standard nav tabs are not shown.
+          if (windowDimensionsService.isWindowNarrow()) {
+            return false;
+          }
+
+          // If i18n hasn't completed, retry after 100ms.
+          if (!checkIfI18NCompleted()) {
+            $timeout(truncateNavbar, 100);
+            return false;
+          }
+
+          // Measured non-overflowed navbar height under 60px via inspector.
+          if (document.querySelector('div.collapse.navbar-collapse')
+            .clientHeight > 60) {
+            for (element in $scope.navElementsVisibilityStatus) {
+              if ($scope.navElementsVisibilityStatus[element]) {
+                // Hide one element, then check again after 50ms.
+                // This gives the browser time to render the visibility change.
+                $scope.navElementsVisibilityStatus[element] = false;
+                // Force a digest cycle to hide element immediately.
+                // Otherwise it would be hidden after the next call.
+                // This is due to setTimeout use in debounce.
+                $scope.$apply();
+                $timeout(truncateNavbar, 50);
+                return false;
+              }
+            }
+          }
+        };
+
+        var truncateNavbarDebounced =
+          oppiaDebouncer.debounce(truncateNavbar, 500);
+
+        // For Chrome, timeout 0 appears to run after i18n.
```

Comment:
What I mean here is that in Chrome, a timeout of 0 appears to run after i18n completes - but in other browsers (Firefox, Edge), a timeout of 0 will still run before i18n. I used a timeout of 0 since it works for at least one browser, and in others the function will check that i18n isn't done and retry after 100ms.

Should I add "For other browsers, the function will wait for i18n."?